src/index.js

```diff
@@ -1,32 +1,79 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
+const xhttp = new XMLHttpRequest();
+
+const fetchDataPromesa = url => {
+  // const fetchData = (url_api, callback) => {
+    return new Promise ((resolve,reject)=>{
+      xhttp.onreadystatechange = function () {
+        if (xhttp.readyState === 4) {
+          if (xhttp.status == 200){
+            var respuesta = JSON.parse(xhttp.responseText);
+            /*"respuesta" es la variable que recibe la solicitud
+               en String y se le parsea la respuesta a JSON*/
+            return resolve(respuesta);
```

Comment:
Para mantener un código limpio, te recomiendo hacer el parse de "JSON.parse()" en el resolve de tu función y no hacer JSON.parse por cada llamado. 

```suggestion
        resolve(JSON.parse(xhttp.responseText));
```